File: scripts/import_all_users.py
import re
from datetime import datetime

# Read the breach file with username:email pairs
breach_data = {}
with open('../user_read_only_context/text_attachments/doxbin-breached-2wmPl.txt', 'r', encoding='utf-8', errors='ignore') as f:
    for line in f:
        line = line.strip()
        if ':' in line:
            parts = line.split(':', 1)
            if len(parts) == 2:
                username, email = parts
                breach_data[username.strip()] = email.strip()

print(f"Loaded {len(breach_data)} user:email pairs from breach file")

# Read the password file with ('username', 'password') tuples
password_data = {}
with open('../user_read_only_context/text_attachments/doxbinpws-BB8Qd.txt', 'r', encoding='utf-8', errors='ignore') as f:
    content = f.read()
    # Extract tuples using regex
    pattern = r"$$'([^']+)',\s*'([^']+)'$$"
    matches = re.findall(pattern, content)
    for username, password in matches:
        # Store the last password for each username (in case of duplicates)
        password_data[username.strip()] = password.strip()

print(f"Loaded {len(password_data)} username:password pairs from password file")

# Generate SQL INSERT statements
sql_statements = []
sql_statements.append("-- Doxbin Users Import")
sql_statements.append("-- Generated: " + datetime.now().isoformat())
sql_statements.append("")

# Combine data from both sources
all_usernames = set(list(breach_data.keys()) + list(password_data.keys()))
print(f"Total unique usernames: {len(all_usernames)}")

for username in sorted(all_usernames):
    email = breach_data.get(username, f"{username}@doxbin.local")
    password = password_data.get(username, "defaultpassword123")
    
    # Escape single quotes for SQL
    username_escaped = username.replace("'", "''")
    email_escaped = email.replace("'", "''")
    password_escaped = password.replace("'", "''")
    
    sql = f"INSERT INTO users (name, email, password, role, email_verified, created_at, updated_at) VALUES ('{username_escaped}', '{email_escaped}', '{password_escaped}', 'member', false, NOW(), NOW()) ON CONFLICT (email) DO NOTHING;"
    sql_statements.append(sql)

# Write to output file
output_file = '003_import_all_doxbin_users.sql'
with open(output_file, 'w', encoding='utf-8') as f:
    f.write('\n'.join(sql_statements))

print(f"\nGenerated {len(sql_statements) - 3} SQL INSERT statements")
print(f"Output written to: {output_file}")
print("\nTo import these users, run this script and then execute the generated SQL file.")

File: scripts/import-all-users.ts
import { neon } from "@neondatabase/serverless"
import { readFileSync } from "fs"
import { join } from "path"

const sql = neon(process.env.DATABASE_URL!)

async function importUsers() {
  console.log("Starting user import...")

  // Read the breach file (username:email)
  const breachPath = join(process.cwd(), "user_read_only_context", "text_attachments", "doxbin-breached-2wmPl.txt")
  const breachContent = readFileSync(breachPath, "utf-8")
  const breachLines = breachContent.split("\n").filter((line) => line.trim())

  // Read the password file (SQL tuples)
  const pwPath = join(process.cwd(), "user_read_only_context", "text_attachments", "doxbinpws-BB8Qd.txt")
  const pwContent = readFileSync(pwPath, "utf-8")
  const pwLines = pwContent.split("\n").filter((line) => line.trim())

  // Parse username:email pairs
  const emailMap = new Map<string, string>()
  for (const line of breachLines) {
    const [username, email] = line.split(":")
    if (username && email) {
      emailMap.set(username.trim(), email.trim())
    }
  }

  // Parse password tuples
  const passwordMap = new Map<string, string>()
  for (const line of pwLines) {
    const match = line.match(/$$'([^']+)',\s*'([^']+)'$$/)
    if (match) {
      const username = match[1].trim()
      const password = match[2].trim()
      passwordMap.set(username, password)
    }
  }

  console.log(`Found ${emailMap.size} users with emails`)
  console.log(`Found ${passwordMap.size} users with passwords`)

  // Combine and insert users
  let insertCount = 0
  const users: Array<{ name: string; email: string; password: string }> = []

  for (const [username, email] of emailMap.entries()) {
    const password = passwordMap.get(username) || "default123"
    users.push({
      name: username,
      email: email,
      password: password,
    })
  }

  // Insert in batches of 100
  const batchSize = 100
  for (let i = 0; i < users.length; i += batchSize) {
    const batch = users.slice(i, i + batchSize)

    try {
      for (const user of batch) {
        await sql`
          INSERT INTO users (name, email, password, role, created_at)
          VALUES (
            ${user.name},
            ${user.email},
            ${user.password},
            'user',
            NOW()
          )
          ON CONFLICT (email) DO NOTHING
        `
        insertCount++
      }

      console.log(`Inserted ${Math.min(i + batchSize, users.length)}/${users.length} users...`)
    } catch (error) {
      console.error(`Error inserting batch starting at ${i}:`, error)
    }
  }

  console.log(`\nImport complete! Inserted ${insertCount} users.`)
}

importUsers().catch(console.error)
